manager/dispatcher/dispatcher_test.go

```diff
@@ -1290,54 +1472,150 @@ func filterDependencies(secrets []*api.Secret, configs []*api.Config, inTasks, n
 			filteredConfigs = append(filteredConfigs, c)
 		}
 	}
-	return filteredSecrets, filteredConfigs
+	return uniquifySecrets(filteredSecrets), uniquifyConfigs(filteredConfigs)
+}
+
+func uniquifySecrets(secrets []*api.Secret) []*api.Secret {
+	uniqueSecrets := make(map[string]struct{})
+	var finalSecrets []*api.Secret
+	for _, secret := range secrets {
+		if _, ok := uniqueSecrets[secret.ID]; !ok {
+			uniqueSecrets[secret.ID] = struct{}{}
+			finalSecrets = append(finalSecrets, secret)
+		}
+	}
+	return finalSecrets
+}
+
+func uniquifyConfigs(configs []*api.Config) []*api.Config {
+	uniqueConfigs := make(map[string]struct{})
+	var finalConfigs []*api.Config
+	for _, config := range configs {
+		if _, ok := uniqueConfigs[config.ID]; !ok {
+			uniqueConfigs[config.ID] = struct{}{}
+			finalConfigs = append(finalConfigs, config)
+		}
+	}
+	return finalConfigs
 }
 
+type taskGeneratorFunc func(t *testing.T, nodeID string) ([]*api.Secret, []*api.Config, []*api.ResourceReference, []*api.Task)
+
 // Creates 1 task for every possible task state, so there are 12 tasks, ID=0-11 inclusive.
 // Creates 1 secret and 1 config for every single task state + 1, so there are 13 secrets, 13 configs, ID=0-12 inclusive
+// Creates 1 secret and 1 config per task by resource reference so there are an additional of each eventually created
 // For each task, the dependencies assigned to it are: secret<i>, secret12, config<i>, config12.
```

Comment:
"For each task, the dependencies assigned to it are: secret<i>, secret12, config<i>, config12, resourceRefSecret<i>, resourceRefConfig<i>"?